src/common/SIRF.py

```diff
@@ -171,6 +171,8 @@ def add(self, other, out=None):
         other: DataContainer
         out:   DataContainer to store the result to.
         '''
+        if not isinstance (other, ( DataContainer , Number, int, float, numpy.float32 )):
```

Comment:
int, float and numpy.float32 are derived from Number, are they not?

Reply:
I suppose `int` and `float` may be, but I wouldn't be sure about `numpy.float32`.

Reply:
>>> import numpy
>>> a = numpy.ones((3,), dtype=numpy.float32)
>>> from numbers import Number
>>> print(isinstance(a[0], Number))
True

Reply:
very well then